wire-ios-notification-engine/Sources/LegacyNotificationSessionLoader.swift

```diff
@@ -0,0 +1,362 @@
+        case buildIsBlacklisted(buildNumber: String)
+
+    }
+
+    private let account: Account
+    private let appContainerURL: URL
+    private let accountDataURL: URL
+    private let appGroupID: String
+    private let buildNumber: String
+    private let sharedUserDefaults: UserDefaults
+    private let minTLSVersion: String?
+
+    private let accountID: UUID
+    private let backendStore: BackendEnvironmentStore
+    private let journal: Journal
+
+    public init(
+        account: Account,
+        appContainerURL: URL,
+        appGroupID: String,
+        buildNumber: String,
+        sharedUserDefaults: UserDefaults,
+        minTLSVersion: String?
+    ) throws {
+        self.account = account
+        self.appContainerURL = appContainerURL
+        self.appGroupID = appGroupID
+        self.buildNumber = buildNumber
+        self.sharedUserDefaults = sharedUserDefaults
+        self.minTLSVersion = minTLSVersion
+
+        self.accountID = account.userIdentifier
+        self.accountDataURL = AccountURLs(root: appContainerURL).accountData
+        self.backendStore = try BackendEnvironmentStore(directory: accountDataURL)
+        self.journal = Journal(
+            userID: accountID,
+            storage: sharedUserDefaults
+        )
+    }
+
+    public func load() async throws -> NotificationSession {
+        // Get the stored environment for this account.
+        let backendEnvironment = try fetchBackendEnvironment()
+
+        // Retrieve proxy credentials if needed.
+        var proxyCredentials: WireNetwork.ProxyCredentials?
+        if let config = backendEnvironment.config.proxyConfig {
+            proxyCredentials = try await fetchProxyCredentials(for: config)
+        }
+
+        // Set up network stack.
+        let preferredAPIVersion = BackendInfo.preferredAPIVersion.flatMap {
+            WireNetwork.APIVersion(rawValue: UInt($0.rawValue))
+        }
+
+        let networkStack = NetworkStack(
+            backendEnvironment: backendEnvironment,
+            minTLSVersion: .minVersionFrom(minTLSVersion),
+            preferredAPIVersion: preferredAPIVersion,
+            proxyCredentials: proxyCredentials
+        )
+
+        let networkServices = try await networkStack.networkServices
+
+        let metadata = try await resolveBackendMetadata(with: networkStack)
+
+        // Set up persistence stack.
+        let coreDataStack = try await setupPersistenceStack(
+            localDomain: metadata.domain,
+            isFederationEnabled: metadata.isFederationEnabled
+        )
+
+        // Return early if needed.
+        guard try await !isBuildBlacklisted(networkService: networkServices.blacklist) else {
+            throw Failure.buildIsBlacklisted(buildNumber: buildNumber)
+        }
+
+        // Return early if needed.
+        guard !shouldEnableSyncV2(metadata: metadata) else {
+            throw Failure.mainAppRequired(message: "sync v2 should be enabled")
+        }
```

Comment:
Should it be this instead ?
```suggestion
        guard shouldEnableSyncV2(metadata: metadata) else {
            throw Failure.mainAppRequired(message: "sync v2 should be enabled")
        }
```

I'm not sure how to read this otherwise, if `shouldEnableSyncV2` is `true` then we throw a failure because sync v2 should be enabled ? seems contradictory

Reply:
Thanks for pointing this out, indeed it was a mistake. I've changed it to return early if sync v2 is enabled.